iphone/Maps/Classes/CustomViews/Login/MWMAuthorizationCommon.mm

```diff
@@ -47,31 +47,46 @@ void AuthorizationStoreCredentials(osm::KeySecret const & keySecret)
   [ud synchronize];
 }
 
-BOOL AuthorizationHaveCredentials()
+BOOL AuthorizationHaveOAuth1Credentials()
 {
   NSUserDefaults * ud = NSUserDefaults.standardUserDefaults;
   NSString * requestToken = [ud stringForKey:kOSMRequestToken];
   NSString * requestSecret = [ud stringForKey:kOSMRequestSecret];
   return requestToken.length && requestSecret.length;
 }
 
+void AuthorizationClearOAuth1Credentials()
+{
+  NSUserDefaults * ud = NSUserDefaults.standardUserDefaults;
+  [ud removeObjectForKey:kOSMRequestToken];
+  [ud removeObjectForKey:kOSMRequestSecret];
+  [ud synchronize];
+}
+
+BOOL AuthorizationHaveCredentials()
+{
+  NSUserDefaults * ud = NSUserDefaults.standardUserDefaults;
+  NSString * oauthToken = [ud stringForKey:kOSMAuthToken];
+  return oauthToken.length;
+}
+
 void AuthorizationClearCredentials()
 {
   NSUserDefaults * ud = NSUserDefaults.standardUserDefaults;
+  [ud removeObjectForKey:kOSMAuthToken];
   [ud removeObjectForKey:kOSMRequestToken];
   [ud removeObjectForKey:kOSMRequestSecret];
   [ud removeObjectForKey:kOSMUserName];
   [ud removeObjectForKey:kOSMChangesetsCount];
   [ud synchronize];
```

Comment:
Can be removed

Reply:
Removed calls to `NSUserDefaults.synchronize` method all over the code